### ISSUE TITLE
Make Panel.ClipToBounds default to false

### ISSUE THREAD

**Issue by mstr2** [AvaloniaUI/Avalonia#2172]
The Panel class overrides the ClipToBounds property and defaults it to true:
https://github.com/AvaloniaUI/Avalonia/blob/5b618dd5da0ca44b21cbe80e2ec972509ebb7f73/src/Avalonia.Controls/Panel.cs#L34

I think this line should be removed for the following reasons:
1. Maintain consistency with WPF, which defaults to false.
2. If you want to opt out of this feature, you need to set it on every level of the visual tree (=> potentially hard to do).
3. On the other hand, if you want to opt in, it is usually sufficient to enable it once at an appropriate level in the visual tree (=> very easy to do).

I encountered this issue while trying to add a focus adorner to a CheckBox control, which is supposed to slightly extend beyond the boundaries of the visual. The fact that all panels clip their children by default makes this very hard to achieve.

**Comment by kekekeks** [comment]
I think it was done mostly for performance reasons so we could track dirty rects more efficiently.

**Comment by kekekeks** [comment]
@grokys

**Comment by grokys** [comment]
Yeah I think I'd be fine with this change, especially if WPF defaults to false. I don't think there was any particular reason true was chosen.
